openfisca_us/variables/gov/hhs/medicaid/eligibility/medicaid_person_type.py

```diff
@@ -25,12 +27,16 @@ def formula(person, period, parameters):
         has_dependents = person.spm_unit.any(
             person.spm_unit.members("age", period) < 19
         )
+        under_21_qualifies = parameters(period).hhs.medicaid.under_21_qualifies_as_child
+        state = person.household("state_code_str", period)
+        state_has_under_21_child_category = under_21_qualifies[state]
         return select(
-            [age == 0, age < 6, age < 19, has_dependents, True],
+            [age == 0, age < 6, age < 19, (age < 21) & state_has_under_21_child_category, has_dependents, True],
```

Comment:
Nice! I didn't understand how this would have worked before

Reply:
Thanks, yeah I did wonder if there was a more elegant solution but this seems to work fine